fixtures/additional-modules/test/index.spec.ts

```diff
@@ -0,0 +1,231 @@
+import assert from "node:assert";
+import childProcess from "node:child_process";
+import { existsSync } from "node:fs";
+import fs from "node:fs/promises";
+import os from "node:os";
+import path from "node:path";
+import {
+	runWranglerDev,
+	wranglerEntryPath,
+} from "../../shared/src/run-wrangler-long-lived";
+import { describe, beforeAll, afterAll, expect, test } from "vitest";
+import { setTimeout } from "node:timers/promises";
+import { fetch } from "undici";
+
+async function getTmpDir() {
+	return fs.mkdtemp(path.join(os.tmpdir(), "wrangler-modules-"));
+}
+
+type WranglerDev = Awaited<ReturnType<typeof runWranglerDev>>;
+function get(worker: WranglerDev, pathname: string) {
+	const url = `http://${worker.ip}:${worker.port}${pathname}`;
+	// Setting the `MF-Original-URL` header will make Miniflare think this is
+	// coming from a `dispatchFetch()` request, meaning it won't return the pretty
+	// error page, and we'll be able to parse errors as JSON.
```

Comment:
It should probably be checking the Accept header to see if `text/html` is there before it sends a pretty error page?

Reply:
Are you saying that Miniflare should do better or this test?